src/java.base/share/classes/java/io/FileInputStream.java

```diff
@@ -131,11 +131,6 @@ public FileInputStream(String name) throws FileNotFoundException {
     @SuppressWarnings("this-escape")
     public FileInputStream(File file) throws FileNotFoundException {
         String name = (file != null ? file.getPath() : null);
-        @SuppressWarnings("removal")
-        SecurityManager security = System.getSecurityManager();
-        if (security != null) {
-            security.checkRead(name);
-        }
         if (name == null) {
             throw new NullPointerException();
         }
```

Comment:
I assume (old) L133-139 can be replaced with `String name = file.getPath()`, the explicit check that name is null goes away with the SM check.